public/components/header.jsx

```diff
@@ -1,20 +1,34 @@
+import { useEffect, useState } from "react";
 import css from "../css/header.module.css";
+import UserGreeting from "./userGreeting";
 
 export default function Header() {
   const currentDate = new Date().toLocaleDateString("en-gb", {
     day: "numeric",
     month: "numeric",
     year: "numeric",
   });
+
   const title = "Pokèmon";
+
+  const [user, setUser] = useState();
+  useEffect(() => {
+    if (typeof window !== "undefined") {
+      const storedUser = localStorage.getItem("user");
+      setUser(storedUser);
+    }
+  });
```

Comment:
no dependency array

Reply:
if there is a dependency array the showing of user will require a reload of the page. (checked with [user], [window], [ ]).

Reply:
why no dependency array?